File: Search and Sort/SelectionSort.py
def selectionSort(array):
    for i in range(len(array)):
        candidate = i
        for j in range(i + 1, len(array)):
            if (array[candidate] > array[j]):
                candidate = j
        array[i], array[candidate] = array[candidate], array[i]

if __name__ == "__main__":
    array = [ 64, 25, 12, 22, 11 ]
    selectionSort(array)
    print(array)

File: Search and Sort/SelectionSort.cpp
#include <cstdio>
#include <algorithm>
using namespace std;

int selectionSort(int array[], int length) {
    for (int i = 0; i < length; i++) {
        int candidate = i;
        for (int j = i + 1; j < length; j++) {
            if (array[candidate] > array[j])
                candidate = j;
        }
        swap(array[i], array[candidate]);
    }
}

int printArray(int array[], int length) {
    for (int i = 0; i < length; i++) {
        printf("%d ", array[i]);
    }
}

int main(int argc, char **argv) {
    int array[] = { 64, 25, 12, 22, 11 };
    int length = sizeof(array) / sizeof(int);

    selectionSort(array, length);

    printArray(array, length);

    return 0;
}